src-clojure/me/tonsky/persistent_sorted_set.cljs

```diff
@@ -287,145 +311,273 @@
     (let [nodes (node-merge-n-split node right)]
       (return-array left (arrays/aget nodes 0) (arrays/aget nodes 1)))))
 
-(deftype Node [keys pointers]
+(defprotocol IStore
+  (store-aux [this storage]))
+
+(defn- node-addresses->array
+  [^js children]
+  (let [children-addresses (map #(gobj/get % "_address") children)]
```

Comment:
Will this work with advanced compilation? I’d like to avoid externs as much as possible. Why not `(.-_address %)`?

Reply:
Thanks! I just changed it to use `(.-_address %)`, there's still the issue with `_address` for advanced compilation in our case, so we added it to externs as a workaround.